tests/cases/fourslash/completionsImport_reExportDefault.ts

```diff
@@ -28,16 +28,6 @@ verify.completions({
             hasAction: true,
             sortText: completion.SortText.AutoImportSuggestions
         },
-        {
-            name: "foo",
-            source: "/a/index",
-            sourceDisplay: "./a",
-            text: "(alias) function foo(): void\nexport foo",
-            kind: "alias",
-            kindModifiers: "export",
-            hasAction: true,
-            sortText: completion.SortText.AutoImportSuggestions
-        },
```

Comment:
So are we now preferring comletions with a longer path? It seems like we've removed the duplicate, but are leaving in a suggestion for a farther module.

Reply:
No. The real gotcha is that the actual value of the `source` property, as long as it does in fact point to _some_ module that exports the thing you want to import, is _completely irrelevant_ in determining the final module specifier we use. When we get completion details, we use that `source` and `name` to find the export, then we skip all aliases and find the symbol at its actual declaration, then we again find _every re-export_ of it and decide which one is best. So literally any module in a chain of re-exporting modules is just as good as any other, making the whole bucketing thing I deleted completely pointless. Notice that `sourceDisplay` on the remaining one (and equivalently, the resulting module specifier after applying the completion) is still `"./a"` (the best and shortest module specifier) even though the `source` property happens to be a longer path.

Reply:
The exception to this “completely irrelevant” claim is when one symbol is re-exported from different ambient modules, as coded into `moduleSymbolsAreDuplicateOrigins`. Any number of re-exporting external modules can be considered together, but each ambient module is unique, since we really have no concept of whether a module named `"ambient-module-one"` or `"ambient-module-two"` is a better place to import from. We treat those as unique, whereas we don’t give you a choice between `"./a"` and `"./a/index"` and `"./a/actual/declaration/module" if the same symbol can be accessed from any of them.